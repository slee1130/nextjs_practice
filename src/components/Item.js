import { Button } from "semantic-ui-react";

export default function Item({ item }) {
  const { image_link, name, price, description } = item;
  return (
    <>
      <div>
        <img src={image_link} alt={name} />
      </div>
      <div>
        <strong>{name}</strong>
        <strong>${price}</strong>
      </div>
      <Button color="orange">Buy</Button>
      <div>
        <p>{description}</p>
      </div>
    </>
  );
  console.log(item);
  return <div>hihi</div>;
}
