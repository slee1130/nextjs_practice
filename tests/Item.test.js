import Item from "../src/components/Item";
import "@testing-library/jest-dom";
import { render, fireEvent, waitFor, screen } from "@testing-library/react";
import styles from "../components/Item.module.css";

describe("check we get all items from items", () => {
  test("check if we have any items", async () => {
    const { debug, container } = render(<Item />);
    expect(container.firstChild).toBeNull();
    debug();
    // expect(item.length > 0).toBeTruty();
  });
  test.only("정상적으로 props를 렌더링 하는지 확인", async () => {
    const mock = {
      image_link: "/test",
      name: "서울우유",
      price: 300,
      description: "우유는 맛있어",
      category: "유제품이다",
      product_type: "유제품",
    };

    const { debug, container, getByText } = render(<Item item={mock} />);
    expect(container).toMatchSnapshot();
    const productImageEl = container.querySelector("img");

    expect(productImageEl.alt).toEqual(mock.name);
    expect(productImageEl).toHaveAttribute("src", mock.image_link);

    const priceDivEl = container.querySelector(`.${styles.info_item}`);
    expect(priceDivEl.firstChild.tagName).toEqual("STRONG");
    expect(priceDivEl.firstChild).toHaveTextContent(mock.name);
  });
});
