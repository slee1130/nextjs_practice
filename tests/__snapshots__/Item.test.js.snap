// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check we get all items from items 정상적으로 props를 렌더링 하는지 확인 1`] = `
<div>
  <div
    class="wrap"
  >
    <div
      class="img_item"
    >
      <img
        alt="서울우유"
        src="/test"
      />
    </div>
  </div>
  <div
    class="info_item"
  >
    <strong
      class="tit_item"
    >
      서울우유
    </strong>
    <strong
      class="num_price"
    >
      $
      300
    </strong>
  </div>
  <span
    class="txt_info"
  >
    유제품이다/
    유제품
  </span>
  <button
    class="ui orange button"
  >
    Buy
  </button>
  <h3
    class="ui header"
  >
    Description
  </h3>
  <p
    style="padding-bottom: 20px; font-size: 18px;"
  >
    우유는 맛있어
  </p>
</div>
`;
